Objects/bytearrayobject.c

```diff
@@ -141,22 +141,26 @@ PyByteArray_FromStringAndSize(const char *bytes, Py_ssize_t size)
     }
 
     new = PyObject_New(PyByteArrayObject, &PyByteArray_Type);
-    if (new == NULL)
+    if (new == NULL) {
         return NULL;
+    }
 
     if (size == 0) {
+        new->ob_bytes_object = NULL;
         new->ob_bytes = NULL;
         alloc = 0;
     }
     else {
         alloc = size + 1;
-        new->ob_bytes = PyMem_Malloc(alloc);
+        new->ob_bytes_object = PyBytes_FromStringAndSize(NULL, alloc);
```

Comment:
@cmaloney 

What do you think about adding a NULL check here?


```suggestion
        new->ob_bytes_object = PyBytes_FromStringAndSize(NULL, alloc);
        if (new->ob_bytes_object == NULL) {
            Py_DECREF(new);
            return NULL;
        }
```

Reply:
:+1: in that `PyBytes_FromStringAndSize` may return null. I'm working on a refactor of this to get the tests passing in general, will look at incorporating / make sure have a fix.